source/server/connection_handler_impl.cc

```diff
@@ -288,5 +310,60 @@ ListenerStats ConnectionHandlerImpl::generateStats(Stats::Scope& scope) {
   return {ALL_LISTENER_STATS(POOL_COUNTER(scope), POOL_GAUGE(scope), POOL_HISTOGRAM(scope))};
 }
 
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerConfig& config)
+    : ActiveUdpListener(parent, parent.dispatcher_.createUdpListener(config.socket(), *this),
+                        config) {}
+
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerPtr&& listener,
+                                                            Network::ListenerConfig& config)
+    : ConnectionHandlerImpl::ActiveListenerBase(parent, std::move(listener), config),
+      udp_listener_(dynamic_cast<Network::UdpListener*>(listener_.get())) {
```

Comment:
similar comment. It would be nice to not have to dynamic_cast here. Maybe a TODO to figure out something better.